lib/storage.js

```diff
@@ -53,6 +53,14 @@ async function getAsset(assetReference, directory, name, disableDownload) {
         const source = new Asset(assetReference, directory, name);
         if (disableDownload) {
             console.log(`Skipping source file download for ${AssetComputeLogUtils.redactUrl(source.url)}`);
+            const protocol = URL.parse(assetReference.url).protocol;
+            if (protocol === DATA_PROTOCOL) {
+                await datauri.download(assetReference, source.path);
+                const preSignedUrl = await datauri.getPreSignedUrl(source.path);
```

Comment:
I would do the existence check of the file `source.path` here (after L58), followed by the verification the size > 0. Since this is where the download is done, this could be the best location to verify if the file was properly created...

Reply:
`datauri.getPreSignedUrl` could throw error for `temporaryCloudStorage.upload / temporaryCloudStorage.generatePresignURL` - we should handle it and also have a test case. This is a common flow and an error here could cause regression. So its good to confirm.